src/zos_files/zowe/zos_files_for_zowe_sdk/files.py

```diff
@@ -163,6 +165,10 @@ def create_data_set(self, dataset_name, options = {}):
         json
         """
 
+        if options.get("like") is None:
+            if options.get("primary") is None or options.get("lrecl") is None:
+                raise KeyError
+
         for opt in ("volser", "unit", "dsorg", "alcunit", 
```

Comment:
I believe that whenever the `like` option is specified in the `dict`, we shouldn't be adding any default values (e.g. `alcunit: TRK`, `recfm: F`)
Maybe we just need to include this `for` loop inside the `if('like') is None` condition 😋

Reply:
Thank you, I will change that!